Maple2.Model/Game/Shop/Shop.cs

```diff
@@ -0,0 +1,49 @@
+﻿using Maple2.Model.Enum;
+using Maple2.PacketLib.Tools;
+using Maple2.Tools;
+using Maple2.Tools.Extensions;
+
+namespace Maple2.Model.Game.Shop;
+
+public class Shop : IByteSerializable {
+    public readonly int Id;
+    public int NpcId;
+    public int CategoryId { get; init; }
+    public string Name  { get; init; }
+    public ShopSkin Skin  { get; init; }
+    public bool HideUnuseable  { get; init; }
+    public bool HideStats  { get; init; }
+    public bool DisableBuyback  { get; init; }
+    public bool OpenWallet  { get; init; }
+    public bool DisplayNew  { get; init; }
+    public bool RandomizeOrder  { get; init; }
+    public bool CanRestock  { get; init; }
+    public long RestockTime  { get; init; }
+    public ShopRestockData RestockData { get; init; }
```

Comment:
should we just make `ShopRestockData` nullable, then you dont need `CanRestock` and instead you can just check `RestockData != null`